### PR TITLE
fix: delete from docarray by doc id

### PR DESCRIPTION
**Pull request by cristianmtr**

### CODE REVIEW

File: jina/types/arrays/document.py [hunk 1]
```diff
@@ -164,11 +166,11 @@ def reverse(self):
             self._docs_proto.reverse()
 
     @cached_property
-    def _docs_map(self):
-        """Returns a doc_id to doc mapping so one can later index a Document using doc_id as string key.
+    def _id_to_index(self):
```
Comment (JoanFM) [inline jina/types/arrays/document.py:169]: are we sure this is to br substituted, isnt it compatible? where is the docsmap used now?

Comment (hanxiao) [reply jina/types/arrays/document.py:169]: `docs_map` is used in `__index__` to enable via-`id`-access https://github.com/jina-ai/jina/blob/master/.github/2.0/cookbooks/Document.md#access-element

Comment (cristianmtr) [reply jina/types/arrays/document.py:169]: I don't understand the question. 

I didn't see where else `_docs_map` was used. The interface is respected, as defined by the existing tests

Comment (JoanFM) [reply jina/types/arrays/document.py:169]: ah ok it is private and is used smartly. Ok good for me

Comment (cristianmtr) [reply jina/types/arrays/document.py:169]: can I get it approved then?

**Review by JoanFM — APPROVED** [review]

**Pull request merged**